File: Spring 2020/RemoveValueFromLinkedList/Java/Main.java
public class Main {
    public static void main(String []args) {
        Node E = new Node(5, null);
        Node D = new Node(4, E);
        Node C = new Node(3, D);
        Node B = new Node(2, C);
        Node A = new Node(1, B);
        Node pointer = A;
        while (pointer != null) {
            System.out.println(pointer.value);
            pointer = pointer.next;
        }
        System.out.println("-----");
        removeNode(A, 5);
        pointer = A;
        while (pointer != null) {
            System.out.println(pointer.value);
            pointer = pointer.next;
        }
    }

    public static void removeNode(Node head, int value) {
        Node previous = null, pointer = head;
        while (pointer != null && pointer.value != value) {
            previous = pointer;
            pointer = pointer.next;
        }

        if (pointer != null && previous != null) previous.next = pointer.next;
        if (previous == null) pointer = null;
    }

    public static class Node {
        public int value;
        public Node next;
    
        public Node(int value, Node next) {
            this.value = value;
            this.next = next;
        }
    }
}

File: Spring 2020/RemoveValueFromLinkedList/Python/main.py
class Node:
    def __init__(self, value, next):
        self.value = value 
        self.next = next 

def remove_node(head, value):
    pointer, previous = head, None
    while pointer != None and pointer.value != value:
        previous = pointer
        pointer = pointer.next

    if (pointer and previous):
        previous.next = pointer.next

E = Node(5, None)
D = Node(4, E)
C = Node(3, D)
B = Node(2, C)
A = Node(1, B)

pointer = A
while pointer:
    print(pointer.value)
    pointer = pointer.next

print("-----")

remove_node(A, 5)
pointer = A
while pointer:
    print(pointer.value)
    pointer = pointer.next
